File: src/main/java/com/natia/secretmod/features/AlertPickups.java
package com.natia.secretmod.features;

import com.natia.secretmod.SecretUtils;
import com.natia.secretmod.core.ItemPickupEvent;
import net.minecraft.client.Minecraft;
import net.minecraft.util.EnumChatFormatting;
import net.minecraft.util.StringUtils;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

import java.util.ArrayList;
import java.util.List;

public class AlertPickups {

    List<String> alertedItems = new ArrayList<>();
    public AlertPickups() {
        /* frags */
        alertedItems.add("Diamante's Handle");
        alertedItems.add("L.A.S.R.'s Eye");
        alertedItems.add("Bigfoot's Lasso");
        alertedItems.add("Jolly Pink Rock");
        /* diana */
        alertedItems.add("Chimera I");
        alertedItems.add("Dwarf Turtle Shelmet");
        alertedItems.add("Antique Remedies");
        alertedItems.add("Minos Relic");
        alertedItems.add("Daedalus Stick");
    }
    private Minecraft mc = Minecraft.getMinecraft();

    @SubscribeEvent
    public void onPickup(ItemPickupEvent event) {
        if (event.getItemDiff().getAmount() < 0) return;
        if (mc.currentScreen != null) return;
        System.out.println("Pickup - " + event.getItemDiff().getDisplayName());
        if (alertedItems.contains(StringUtils.stripControlCodes(event.getItemDiff().getDisplayName()))) {
            SecretUtils.playLoudSound("random.orb", 0.5f);
            SecretUtils.sendMessage(EnumChatFormatting.GOLD + "" + EnumChatFormatting.BOLD + "RARE DROP! " + EnumChatFormatting.RESET + event.getItemDiff().getDisplayName());
        }
    }

}

File: src/main/kotlin/natia/skytweaks/features/AlertPickups.kt
package natia.skytweaks.features

import mixin.natia.skytweaks.SkyTweaksConfig
import natia.skytweaks.core.ItemPickupEvent
import natia.skytweaks.SecretUtils
import natia.skytweaks.SkyTweaks
import net.minecraft.client.Minecraft
import net.minecraft.util.EnumChatFormatting
import net.minecraft.util.StringUtils

import java.util.ArrayList

class AlertPickups {

    internal var alertedItems: MutableList<String> = ArrayList()
    private val mc = Minecraft.getMinecraft()

    init {
        /* frags */
        alertedItems.add("Diamante's Handle")
        alertedItems.add("L.A.S.R.'s Eye")
        alertedItems.add("Bigfoot's Lasso")
        alertedItems.add("Jolly Pink Rock")
        /* diana */
        alertedItems.add("Chimera I")
        alertedItems.add("Dwarf Turtle Shelmet")
        alertedItems.add("Antique Remedies")
        alertedItems.add("Minos Relic")
        alertedItems.add("Daedalus Stick")
    }

    fun pickup(event: ItemPickupEvent) {
        if (event.getItemDiff().amount < 0) return
        if (mc.currentScreen != null) return
        if (!SkyTweaksConfig.alertRareDrops) return

        SkyTweaks.LOGGER.info("Pickup - " + event.getItemDiff().displayName)
        if (alertedItems.contains(StringUtils.stripControlCodes(event.getItemDiff().displayName))) {
            SecretUtils.playLoudSound("random.orb", 0.5f)
            SecretUtils.sendMessage(EnumChatFormatting.GOLD.toString() + "" + EnumChatFormatting.BOLD + "RARE DROP! " + EnumChatFormatting.RESET + event.getItemDiff().displayName)
        }
    }

    companion object {
        val instance = AlertPickups()
    }

}
